src/fe/fe.C

```diff
@@ -297,17 +297,32 @@ void FE<Dim,T>::reinit(const Elem * elem,
       {
         if (T != LAGRANGE)
           nonlagrange_dual_warning();
-
-        // In order for the matrices for the biorthogonality condition to be
-        // non-singular, the dual basis coefficients must be computed when the
-        // primal shape functions are evaluated with a quadrature rule. *But* a
-        // user may be reiniting with integration points from a mortar segment,
-        // which is valid, so a simple `if (!pts)` check is not
-        // appropriate. We're just gonna have to trust the user on this one. If
-        // they "screw up" we'll throw an exception from the LU decomposition,
-        // and they can choose to handle it or not
-        this->compute_dual_shape_coeffs();
-
+        if (this->calculate_dual_coeff)
+        {
+          // In order for the matrices for the biorthogonality condition to be
+          // non-singular, the dual basis coefficients must be computed when the
+          // primal shape functions are evaluated with a quadrature rule. *But* a
+          // user may be reiniting with integration points from a mortar segment,
+          // which is valid, so a simple `if (!pts)` check is not
+          // appropriate. We're just gonna have to trust the user on this one. If
+          // they "screw up" we'll throw an exception from the LU decomposition,
+          // and they can choose to handle it or not
+
+          // instead of using the customized qrule from mortar segments,
+          // we need the default qrule for computing the dual coefficients
+          FEType default_fe_type(this->get_order(), T);
+          QGauss default_qrule(elem->dim(), default_fe_type.default_quadrature_order());
+          // In preparation of computing dual_coeff, we compute the default shape
+          // function values and save these in dual_phi (instead of declaring a
+          // new container). The TRUE dual_phi values are computed in
+          // compute_dual_shape_functions()
+          all_shapes(elem, default_fe_type.order, default_qrule.get_points(), this->dual_phi);
```

Comment:
Thanks @jwpeterson !

Sounds like the `default_qrule` that is created here has an empty `_points` and is causing the segfault in `all_shapes` later..

So I may need an additional init step to resize `_points` in `default_qrule`, e.g. `default_qrule.init(elem->type(), elem->p_level());`?

Reply:
Yes, I think you are correct. We don't normally manually `init()` a quadrature rule in user code so it's easy to forget this.